1-js/02-first-steps/12-nullish-coalescing-operator/article.md

```diff
@@ -1,166 +1,169 @@
 
-So if we'd like to choose a value with `??` in an expression with other operators, consider adding parentheses:
+Então, se quiser selecionar um valor com `??` em uma expressão com outros operadores, considere o uso de parênteses:
 
 ```js run
-let height = null;
-let width = null;
+let altura = null;
+let largura = null;
 
-// important: use parentheses
-let area = (height ?? 100) * (width ?? 50);
+// importante: use parênteses
+let area = (altura ?? 100) * (largura ?? 50);
 
 alert(area); // 5000
 ```
 
-Otherwise, if we omit parentheses, then as `*` has the higher precedence than `??`, it would execute first, leading to incorrect results.
+Caso contrário, se omitirmos os parênteses, como `*` tem maior precedência que `??`, ele será executado primeiro, levando a resultados incorretos.
 
 ```js
-// without parentheses
-let area = height ?? 100 * width ?? 50;
+// sem parênteses
+let area = altura ?? 100 * largura ?? 50;
 
-// ...works the same as this (probably not what we want):
-let area = height ?? (100 * width) ?? 50;
+// ...funciona desta forma (provavelmente não como gostaríamos):
+let area = altura ?? (100 * largura) ?? 50;
 ```
 
-### Using ?? with && or ||
+### Usando ?? com && ou ||
 
-Due to safety reasons, JavaScript forbids using `??` together with `&&` and `||` operators, unless the precedence is explicitly specified with parentheses.
+Por razões de segurança, o JavaScript proibe o uso de `??` juntamente com os operadores `&&` e `||`, a menos que a precedência seja explicitamente especificada usando parênteses.
 
-The code below triggers a syntax error:
+O código abaixo dispara um erro de sintaxe:
 
 ```js run
-let x = 1 && 2 ?? 3; // Syntax error
+let x = 1 && 2 ?? 3; // Erro de sintaxe
 ```
 
-The limitation is surely debatable, but it was added to the language specification with the purpose to avoid programming mistakes, when people start to switch to `??` from `||`.
+A limitação é certamente discutível, mas foi incluída na especificação da linguagem com o propósito de evitar erros de programação, quando as pessoas começaram a usar `??` em vez de `||`.
 
-Use explicit parentheses to work around it:
+Use parênteses explícitos para corrigí-la:
 
 ```js run
 *!*
-let x = (1 && 2) ?? 3; // Works
+let x = (1 && 2) ?? 3; // Funciona
 */!*
 
 alert(x); // 2
 ```
 
-## Summary
+## Resumo
 
-- The nullish coalescing operator `??` provides a short way to choose the first "defined" value from a list.
+- O operador de coalescência nula `??` disponibiliza uma sintaxe curta para obter um valor "definido" em uma lista.
 
-    It's used to assign default values to variables:
+    É usado para atribuir valores a variáveis:
 
     ```js
-    // set height=100, if height is null or undefined
-    height = height ?? 100;
+    // grava altura=100, se altura é null ou undefined
```

Comment:
Hi, I don't know the usual Brazilian word, here could it be?
```suggestion
    // guarda altura=100, se altura é null ou undefined
```

Reply:
Both words are correct, but "grava" is better.